Bucket/Blob.ts

```diff
@@ -0,0 +1,39 @@
+import * as platform from "@cloudflare/workers-types"
+import { Bucket } from "./Bucket"
+
+export class Blob<M extends Record<string, string> | undefined = undefined> extends Bucket<platform.Blob, M> {
+	async store(key: string, value: platform.Blob, meta: M): Promise<Blob.Result<M> | undefined> {
+		await this.backend.put(key, value, { customMetadata: meta })
+		return { value, meta }
+	}
+	async get(key: string): Promise<Blob.Result<M> | undefined> {
+		const object = await this.backend.get(key)
+		const value = await object?.blob()
+		const meta = object?.customMetadata as M
+		return value && { value, meta }
+	}
+	async remove(key: string | string[]): Promise<void> {
+		this.backend.delete(key)
+	}
+	async list(options?: Bucket.ListOptions): Promise<(Partial<Blob.Result<M>> & { key: string })[]> {
+		const list = (await this.backend.list(options)).objects.map(i => ({
+			key: i.key,
+			meta: i.customMetadata as M,
+		}))
+		return options?.values
+			? await Promise.all(list.map(i => ({ ...i, value: this.get(i.key).then(v => v?.value) })))
```

Comment:
the promise behind the `value` property should be awaited?
running this in the javascript console results in the value being promises in the objects
```js
await Promise.all([1, 2, 3].map(async (value) => ({value: new Promise(resolve => resolve(value))}))) 
```
![image](https://github.com/utily/cloudly-storage/assets/25643315/61c9c668-079f-4528-aadf-3cdda0b42f8c)

Reply:
Fixed.